templates/fine-tune-stable-diffusion/README.ipynb

```diff
@@ -0,0 +1,392 @@
+    "\n",
+    "from train_dreambooth_lora_sdxl import main\n",
+    "from utils import download_from_cloud, upload_to_cloud\n",
+    "\n",
+    "\n",
+    "# Set environment variables across the entire cluster.\n",
+    "WANDB_API_KEY = os.environ.get(\"WANDB_API_KEY\")\n",
+    "if WANDB_API_KEY:\n",
+    "    ray.init(\n",
+    "        runtime_env={\"env_vars\": {\"WANDB_API_KEY\": WANDB_API_KEY}},\n",
+    "        ignore_reinit_error=True,\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "def train_fn_per_worker(config: dict):\n",
+    "    download_from_cloud(cloud_uri=DATA_CLOUD_PATH, local_path=SUBJECT_IMAGES_PATH)\n",
+    "\n",
+    "    # See train_dreambooth_lora_sdxl.py for all of the training details.\n",
+    "    final_checkpoint_path = main(config[\"args\"])\n",
+    "\n",
+    "    # Upload final checkpoint to cloud. (Only the rank 0 worker will return a path here.)\n",
+    "    if final_checkpoint_path is not None:\n",
+    "        upload_to_cloud(local_path=final_checkpoint_path, cloud_uri=MODEL_CHECKPOINT_PATH)\n",
+    "        print(\"Final checkpoint has been uploaded to: \", MODEL_CHECKPOINT_PATH)\n",
+    "\n",
+    "\n",
+    "trainer = TorchTrainer(\n",
+    "    train_fn_per_worker,\n",
+    "    # Pass command line arguments to the `config` dict of the `train_fn_per_worker`\n",
+    "    train_loop_config={\"args\": TRAINING_ARGS},\n",
+    "    scaling_config=ray.train.ScalingConfig(\n",
+    "        # Do data parallel training with A10G GPU workers\n",
+    "        num_workers=4, use_gpu=True, accelerator_type=\"A10G\"\n",
+    "    ),\n",
+    ")\n",
+    "\n",
+    "# Launch the training.\n",
+    "trainer.fit()\n",
+    "print(\"Finished fine-tuning!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Generate some images with your fine-tuned model!\n",
+    "\n",
+    "Finally, let's generate some images!\n",
+    "\n",
+    "We'll launch 2 remote GPU tasks to generate images from the `PROMPTS` we defined earlier, one using just the base model and one that loads our fine-tuned LoRA weights. Let's compare them to see the results of fine-tuning!\n",
+    "\n",
+    "Note: If your cluster has already scaled down from the training job due to the workers being idle, then this step might take a little longer to relaunch new GPU workers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ray\n",
+    "from utils import generate\n",
+    "\n",
+    "[base_model_images, finetuned_images] = ray.get([\n",
+    "    generate.remote(prompts=PROMPTS, args=TRAINING_ARGS),\n",
+    "    generate.remote(\n",
+    "        prompts=PROMPTS, args=TRAINING_ARGS, model_checkpoint_path=MODEL_CHECKPOINT_PATH\n",
+    "    )\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Images generated with the base model"
```

Comment:
This is a little confusing, how about just having a cell to display the finedtuned images, and a comment you can uncomment on how to display images for the base model?

Users will not read carefully and might think these are the fine-tuned images.

Reply:
Good point, I'll comment out the base model generation